dvc/repo/ls.py

```diff
@@ -65,15 +94,17 @@ def _ls_outs_repo(repo, target_path_info, recursive=None):
         return []
 
     if recursive:
-        return [out.path_info for out in outs]
+        return map(lambda out: get_fs_node(out.path_info, True), outs)
 
     def get_top_part(path_info):
         relpath = path_info.relpath(target_path_info)
         if relpath.parts:
-            return PathInfo(target_path_info, relpath.parts[0])
-        return path_info
+            isout = len(relpath.parts) == 1
+            return (PathInfo(target_path_info, relpath.parts[0]), isout)
+        return (path_info, True)
 
-    return list({get_top_part(out.path_info) for out in outs})
+    path_infos = {get_top_part(out.path_info) for out in outs}
+    return map(lambda args: get_fs_node(*args), path_infos)
```

Comment:
`map(lambda ...)` is kind of awkward, also less efficient than a list comprehension.